frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -627,33 +639,50 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
                 <StackItem>
                   {source.requiresVolumeMode() && (
                     <FormRow title={t('kubevirt-plugin~Volume Mode')} fieldId={asId('volume-mode')}>
-                      <FormSelect
-                        onChange={(vMode) => setVolumeMode(VolumeMode.fromString(vMode))}
-                        value={asFormSelectValue(volumeMode?.getValue())}
-                        id={asId('volume-mode')}
-                        isDisabled={isDisabled('volumeMode') || isStorageClassDataLoading}
-                      >
-                        <FormSelectPlaceholderOption
-                          isDisabled={inProgress}
-                          placeholder={t('kubevirt-plugin~--- Select Volume Mode ---')}
-                        />
-                        {VolumeMode.getAll().map((v) => (
-                          <FormSelectOption
-                            key={v.getValue()}
-                            value={v.getValue()}
-                            label={v.toString().concat(
-                              v.getValue() !== defaultVolumeMode.getValue() && isScModesKnown
-                                ? t(
-                                    'kubevirt-plugin~ - Not recommended for {{storageClassName}} storage class',
-                                    {
-                                      storageClassName,
-                                    },
-                                  )
-                                : '',
-                            )}
-                          />
-                        ))}
-                      </FormSelect>
+                      <Stack hasGutter>
```

Comment:
+1 this is a nice solution for disk modals, but it is not enough for the wizard. Template disks are not required to go through disk modals and such volumeMode mistakes will not be catched. Are you planning to fix it here or in the followup?

Reply:
it's a solid point. added a custom hook for this situation.

Reply:
Can we please move this to an updater? The hook works nicely in this scenario, but it poses following issues.

1. Consistency: we try to have all logic in the updaters so we can have one place to look at for changes in the state. If we have the logic scattered everywhere it would become very hard to naviagate. Simple hooks work great for simple things, but if we try to use this approach for the whole wizard it would become a mess.

2. Optimization: this hook is called needlessly on any redux change and when it fires the update it fires the updaters  twice.

Also this logic should target all DV and not just a rootstorage (+ it isn't a good practice to target it by name)